### PR TITLE
cereal 1.1.2

### PR DESCRIPTION
**Pull request by guoxiao**
Add test block

### CODE REVIEW

**Comment by bfontaine** [comment]
The test fails on ML:

```
Testing cereal
==> /usr/bin/clang -std=c++11 -lc++ -o test test.cpp
In file included from test.cpp:1:
In file included from /usr/local/include/cereal/types/unordered_map.hpp:33:
/usr/local/include/cereal/cereal.hpp:32:10: fatal error: 'type_traits' file not found
#include <type_traits>
         ^
1 error generated.
Error: cereal: failed
```

**Comment by guoxiao** [comment]
Should we add a `depends_on :macos => :mavericks`?

**Comment by xu-cheng** [comment]
Try to add `ENV.cxx11` inside `install` method.

File: Library/Formula/cereal.rb [hunk 1]
```diff
@@ -16,8 +16,12 @@ class Cereal < Formula
   option "with-tests", "Build and run the test suite"
 
   depends_on "cmake" => :build if build.with? "tests"
+  depends_on :macos => :mavericks
```
Comment (bfontaine) [inline Library/Formula/cereal.rb]: Please remove this one to see if the `ENV.cxx11` fixed the issue on ML.

Comment (guoxiao) [inline Library/Formula/cereal.rb]: I've tried that before adding this line and it didn't solve the issue of ML.

Comment (bfontaine) [inline Library/Formula/cereal.rb]: Could you try with `Env.cxx11` inside the `test do` block?

Comment (guoxiao) [inline Library/Formula/cereal.rb]: That's exactly what I have tried. :D

Comment (bfontaine) [inline Library/Formula/cereal.rb]: cc @xu-cheng

Comment (xu-cheng) [inline Library/Formula/cereal.rb]: `ENV.cxx11` in the test won't work as the CFLAGS are never used. Try to compile the test file in this way:

```ruby
system ENV.cc, "-std=c++11", "-stdlib=libc++", "-lc++", "-o", "test", "test.cpp"
```

**Comment by guoxiao** [comment]
It seems to be working well on ML now.

**Comment by xu-cheng** [comment]
Thanks for the pull request! 🎉 Homebrew depends on contributions from community members like you and we're grateful for your support.

**Pull request closed without merging**